common/playlist.c

```diff
@@ -195,6 +200,68 @@ void playlist_unshuffle(struct playlist *pl)
     playlist_update_indexes(pl, 0, -1);
 }
 
+void playlist_reorder(struct playlist *pl, const int *indexes, int num_indexes)
+{
+    int current_pl_index = pl->current ? pl->current->pl_index : -1;
+
+    // Mark all entries removed.
+    for (int i = 0; i < pl->num_entries; i++)
+        pl->entries[i]->pl_index = -1;
+
+    int n = 0;
+
+    // Assign indexes.
+    for (int i = 0; i < num_indexes; i++) {
+        int index = indexes[i];
+        struct playlist_entry *e = playlist_entry_from_index(pl, index);
+
+        if (e && e->pl_index < 0)
+            e->pl_index = n++;
+    }
+
+    if (pl->current && pl->current->pl_index < 0) {
+        pl->current = NULL;
+        pl->current_was_replaced = true;
+
+        // Find next entry that is not removed.
```

Comment:
What if all entries after current were removed? Should we pick last not removed one or?